holmes/core/openai_formatting.py

```
@@ -68,6 +68,68 @@ def _is_tool_strict_compatible(tool_parameters: dict) -> bool:
 
 
+def _is_azure_compatible(tool_parameters: dict) -> bool:
+    """Check if a tool's parameters are compatible with Azure OpenAI strict schema.
+
+    Azure OpenAI requires that 'required' only includes keys present in 'properties'.
+    Tools with dynamic object arguments (additionalProperties with a schema) may have
+    'required' keys that don't match 'properties', causing Azure to reject the schema.
+
+    Returns True if compatible, False if the tool should be filtered out for Azure.
+    """
+    for param_name, param_attributes in tool_parameters.items():
```

Ruff (0.16.0)

[warning] Loop control variable `param_name` not used within loop body

Rename unused `param_name` to `_param_name`

(B007)